src/sentry/api/endpoints/organization_event_details.py

```diff
@@ -40,60 +41,65 @@ def get(self, request, organization, project_slug, event_id):
         event_slug = u"{}:{}".format(project.slug, event_id)
         snuba_args["conditions"].extend(get_reference_event_conditions(snuba_args, event_slug))
 
+        filter = self._get_filter(snuba_args)
+
         data = serialize(event)
-        data["nextEventID"] = self.next_event_id(snuba_args, event)
-        data["previousEventID"] = self.prev_event_id(snuba_args, event)
-        data["oldestEventID"] = self.oldest_event_id(snuba_args, event)
-        data["latestEventID"] = self.latest_event_id(snuba_args, event)
+        data["nextEventID"] = self.next_event_id(event, filter=filter)
+        data["previousEventID"] = self.prev_event_id(event, filter=filter)
+        data["oldestEventID"] = self.oldest_event_id(event, filter=filter)
+        data["latestEventID"] = self.latest_event_id(event, filter=filter)
         data["projectSlug"] = project_slug
 
         return Response(data)
 
-    def next_event_id(self, snuba_args, event):
+    def next_event_id(self, event, filter):
         """
         Returns the next event ID if there is a subsequent event matching the
         conditions provided. Ignores the project_id.
         """
-        next_event = eventstore.get_next_event_id(event, filter=self._get_filter(snuba_args))
+        next_event = eventstore.get_next_event_id(event, filter=filter)
 
         if next_event:
             return next_event[1]
 
-    def prev_event_id(self, snuba_args, event):
+    def prev_event_id(self, event, filter):
         """
         Returns the previous event ID if there is a previous event matching the
         conditions provided. Ignores the project_id.
         """
-        prev_event = eventstore.get_prev_event_id(event, filter=self._get_filter(snuba_args))
+        prev_event = eventstore.get_prev_event_id(event, filter=filter)
 
         if prev_event:
             return prev_event[1]
 
-    def latest_event_id(self, snuba_args, event):
+    def latest_event_id(self, event, filter):
         """
         Returns the latest event ID if there is a newer event matching the
         conditions provided
         """
-        latest_event = eventstore.get_latest_event_id(event, filter=self._get_filter(snuba_args))
+        latest_event = eventstore.get_latest_event_id(event, filter=filter)
 
         if latest_event:
             return latest_event[1]
 
-    def oldest_event_id(self, snuba_args, event):
+    def oldest_event_id(self, event, filter):
         """
         Returns the oldest event ID if there is a subsequent event matching the
         conditions provided
         """
-        oldest_event = eventstore.get_earliest_event_id(event, filter=self._get_filter(snuba_args))
+        oldest_event = eventstore.get_earliest_event_id(event, filter=filter)
 
         if oldest_event:
             return oldest_event[1]
 
     def _get_filter(self, snuba_args):
+        dataset = snuba.detect_dataset(snuba_args, aliased_conditions=True)
```

Comment:
I'm planning on moving/removing the need to pass `aliased_conditions` in an effort to reduce the number of places/layers we convert from a 'public' field name to the internal ones. Right now that happens in `api.event_search` for conditions and in `snuba.transform_aliases_and_query()`, `snuba.dataset_query()` and `eventstore` methods for selected fields, order and group by.